pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -122,7 +132,26 @@ type PlayerSelector struct {
 	MaxAvailable int64 `json:"maxAvailable,omitempty"`
 }
 
-// ApplyDefaults applies default values to the PlayerSelector
+// CounterSelector is the filter options for a GameServer based on the count and/or available capacity.
+// Undefined or 0 for MaxCount or MaxAvailable means unlimited maximum.
+type CounterSelector struct {
+	MinCount     int64 `json:"minCount"`
+	MaxCount     int64 `json:"maxCount"`
+	MinAvailable int64 `json:"minAvailable"`
+	MaxAvailable int64 `json:"maxAvailable"`
+}
+
+// ListSelector is the filter options for a GameServer based on List available capacity and/or the
+// existence of a value in a List.
+// Undefined or 0 for MaxAvailable means unlimited maximum.
+// Undefined or "" for ValueInList means ignore field. ValueInList is case insensitive.
```

Comment:
why case insensitive? Did we called out somewhere the values in a list are case insensitive?

Reply:
That's a good point. I recommend always defaulting to a more explicit system than explicit. If people want case insensitivity, then they can lowercase all their content before adding/checking/removing it 👍🏻

Reply:
Makes sense, I'll change it to do a direct equality check.